service.go

```diff
@@ -120,6 +123,8 @@ func (s *Service) NewSubnet(mask net.IPMask) (net.IPNet, error) {
 		return net.IPNet{}, microerror.Mask(err)
 	}
 
+	existingSubnets = append(existingSubnets, s.allocatedSubnets...)
```

Comment:
I like the implementation :D

Reply:
Yeah, this was much easier than I expected :D